extjsdk/src/main/java/io/vantiq/extjsdk/ExtensionWebSocketListener.java

```diff
@@ -370,6 +387,19 @@ else if (msg.get("op").equals(ExtensionServiceMessage.OP_QUERY)) {
                         log.warn("Query received with no handler set");
                     }
                 }
+                else if (msg.get("op").equals(ExtensionServiceMessage.OP_RECONNECT_REQUIRED)) {
+                    if (this.reconnectHandler != null) {
+                        this.reconnectHandler.handleMessage(msg);
+                    }
+                    else if (!client.autoReconnect) {
+                        log.warn("Reconnect received with no handler set and no autoconnect. Can no longer "
+                                + "communicate with source '" + client.getSourceName() + "'");
+                    }
+                    if (client.autoReconnect) {
+                        log.info("Automatically attempting to reconnect to source '" + client.getSourceName() + "'");
+                    }
+                    client.sourceReconnection();
```

Comment:
Is there a reason that client.sourceReconnection() is done outside the previous if statement.  I can see that sourceReconnection() checks for whether it should do a reconnection or not, but the name seems wrong if its really doing more than reconnection (like setting status that its now disconnected or something).  Otherwise, either I'd put the sourceReconnection() inside the if, or move the logging to inside that method.

Reply:
You're right about the naming. What it does right now is reset the Future that tells whether the source is connected, and reconnects if told to. I'll separate the two parts out for clarity.